### PR TITLE
Add method to get joint limits from control board

### PR DESCRIPTION
**Pull request by LoreMoretti**
This PR add a member function to get the joint limits of the actuated joints. Also the related python bindings are added.

### CODE REVIEW

File: src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpRobotControl.h [hunk 1]
```diff
@@ -158,6 +158,13 @@ class YarpRobotControl : public IRobotControl
      */
     std::vector<std::string> getJointList() const final;
 
+    /**
+     * Get the actuated joints limits.
+     * @param lowerLimits vector to be filled with the lower limits of the actuated joints.
+     * @param upperLimits vector to be filled with the upper limits of the actuated joints.
+     */
+    void getJointLimits(Eigen::VectorXd& lowerLimits, Eigen::VectorXd& upperLimits) const final;
```
Comment (GiulioRomualdi) [inline src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpRobotControl.h:166]:
```suggestion
    void getJointLimits(Eigen::Ref<Eigen::VectorXd> lowerLimits, Eigen::Ref<Eigen::VectorXd> upperLimits) const final;
```

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp [hunk 2]
```diff
@@ -820,3 +828,14 @@ bool YarpRobotControl::isValid() const
...
+void YarpRobotControl::getJointLimits(Eigen::VectorXd& lowerLimits,
+                                      Eigen::VectorXd& upperLimits) const
+{
+    lowerLimits.resize(m_pimpl->actuatedDOFs);
+    upperLimits.resize(m_pimpl->actuatedDOFs);
+    for (int i = 0; i < m_pimpl->actuatedDOFs; i++)
+    {
+        m_pimpl->controlLimitsInterface->getLimits(i, &lowerLimits[i], &upperLimits[i]);
+    }
+}
```
Comment (GiulioRomualdi) [inline src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:841]:
```suggestion
}

```

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp [hunk 3]
```diff
@@ -820,3 +828,14 @@ bool YarpRobotControl::isValid() const
 {
     return m_pimpl->robotDevice != nullptr;
 }
+
+void YarpRobotControl::getJointLimits(Eigen::VectorXd& lowerLimits,
+                                      Eigen::VectorXd& upperLimits) const
+{
+    lowerLimits.resize(m_pimpl->actuatedDOFs);
+    upperLimits.resize(m_pimpl->actuatedDOFs);
```
Comment (GiulioRomualdi) [inline src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:836]: Using the eigen::Ref you cannot resize it anymore so you should check that the size is correct.

**Review by GiulioRomualdi — CHANGES_REQUESTED** [review]

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp (hunk 3 above)
Comment (LoreMoretti) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:836]: Got it. Indeed this is why I was not using eigen::Ref.

How do you suggest to do the check?

Just an assert, like:

```
assert(lowerLimits.size() == m_pimpl->actuatedDOFs)
assert(upperLimits.size() == m_pimpl->actuatedDOFs)
```

Or you prefer to change also the return type from `void` to `bool` and raise an error if size do not match?

Comment (GiulioRomualdi) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:836]: I would return a bool

Comment (LoreMoretti) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp]: done with https://github.com/ami-iit/bipedal-locomotion-framework/pull/868/commits/a2809e6c8ec68a39c3c4b80b322bd4353560420a

File: src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpRobotControl.h (hunk 1 above)
Comment (LoreMoretti) [reply src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpRobotControl.h]: done with https://github.com/ami-iit/bipedal-locomotion-framework/pull/868/commits/a2809e6c8ec68a39c3c4b80b322bd4353560420a

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**